src/Ticket.php

```diff
@@ -936,9 +936,11 @@ public function prepareInputForUpdate($input)
         // automatic recalculate if user changes urgency or technician change impact
         $canpriority               = Session::haveRight(self::$rightname, self::CHANGEPRIORITY);
         if (
-            (isset($input['urgency']) && $input['urgency'] != $this->fields['urgency'])
-            || (isset($input['impact']) && $input['impact'] != $this->fields['impact'])
-            && ($canpriority && !isset($input['priority']) || !$canpriority)
+            (
+                (isset($input['urgency']) && $input['urgency'] != $this->fields['urgency'])
+                || (isset($input['impact']) && $input['impact'] != $this->fields['impact'])
+            )
+            && (!$canpriority || !isset($input['priority']) || $input['priority'] == $this->fields['priority'])
```

Comment:
User must be able to define manually the priority.

```suggestion
            && (!$canpriority || !isset($input['priority']))
```